### DISCUSSION TITLE
How to get variable in script edge

### DISCUSSION THREAD

**Discussion by muflihfq** [adonisjs/core#1308] [Help] [answered]
I have a controller which passing some data to views

`return view.render('home', { threads: threads })`

I could read data from the tag HTML in edge, but I couldn't read the data in script tag which I need to use jquery

`console.log(threads)`

this code, throw an error, threads is not defined

**Comment by thetutlage** [comment]
Coz script tags are executed in browser and edge templates are rendered on the server.

**Comment by McSneaky** [comment]
You kind of can hack it to work tho
In your Edge template
```html
<script>
window.edgeVariables = {
  threads: {{ threads }}
}
</script>
```
And then use with `console.log(window.edgeVariables.threads)`
It is kind of bad practise to pollute window object too much, but if you add one container which contains all the rest it won't be too bad

I haven't tested the code in example above, but something really similar should work

**Comment by ammezie** [comment]
If you're using inline script on the edge view, you can do:

```ja
<script>
  console.log({{ threads }})
</script>
```

**Comment by thetutlage** [comment, marked as answer]
You will have to stringify the object or array when sending it over the wire. Even though the code appears to be in the same file. The data is going over HTTP and you cannot send native data types via HTTP.

I personally won't recommend using the following code, since you have blurry understanding of the boundaries and you will run into more issues.

```html
  <script type="text/javascript">
    var threads = {{{ JSON.stringify(threads) }}}
    console.log(threads)
  </script>
```

**Reply by yogip07** [reply]
what different between two and three curly brackets?

**Reply by GuillermoFarias** [reply]
> 
https://docs.adonisjs.com/guides/views/templating-syntax#the-output-will-be

**Comment by ardhikarn** [comment]
var threads= {{{ toJSON(threads)}}}
console.log(threads)
